### PR TITLE
Create simpleQueue for use with low quantities of items such as immed…

### PR DESCRIPTION
**Pull request by mcstover**
…iately called analytics events. Implement the queue for any anaylytics events that might be fired prior to the snowplow analtyics library being initialized. Fire any queued events when everything is ready.

### CODE REVIEW

File: src/plugins/kv-analytics-plugin.js [hunk 1]
```diff
@@ -96,44 +98,69 @@ export default Vue => {
...
-
-				// Snowplow API
-				/* eslint-disable max-len */
-				// https://docs.snowplowanalytics.com/docs/collecting-data/collecting-from-own-applications/javascript-tracker/tracking-specific-events/#tracking-custom-structured-events
-				// https://docs.snowplowanalytics.com/docs/collecting-data/collecting-from-own-applications/javascript-tracker/tracking-specific-events/#callback-after-track-2-15-0
-				/* eslint-eable max-len */
-				// snowplow('trackStructEvent', 'category', 'action', 'label', 'property', 'value', context, timestamp, afterTrack);
-				window.snowplow(
-					'trackStructEvent',
+				kvActions.trackSnowplowEvent({
```
Comment (mcstover) [inline src/plugins/kv-analytics-plugin.js:101]: Created an independent event for this for use when re-playing events. While I've focused on snowplow for now we'll probably do the same for ga at some point.

**Comment by mcstover** [comment]
@emuvente Verified I'm getting missed events in the vm including our EXP events on lend by category.

**Review by emuvente — APPROVED** [review]
Awesome!

File: src/util/simpleQueue.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	 */
+	constructor() {
+		this.items = [];
+	}
+
+	/**
+	 * Add an item to the queue
+	 */
+	add(item) {
+		this.items.push(item);
```
[the remark this answers is not in the document]
Comment (mcstover) [reply src/util/simpleQueue.js:17]: I was thinking more like 100 items max as we really shouldn't have too many per page though I suppose they could pile up across in-app page changes...

Do you think it's better to block new items or start removing items and adding the new ones...

[the remark this answers is not in the document]
Comment (emuvente) [reply src/util/simpleQueue.js:22]: I think just blocking the addition of new items makes sense. That way it's not wasting cycles modifying the array

**Pull request merged**